### ISSUE TITLE
Attacking doesn't end when I've killed all of the enemy's bots

### ISSUE THREAD

**Issue by mlindgren** [Adam13531/BotLand#52]
**Steps to reproduce**
Set up some artillery bots which will only attack with artillery or move right. If they kill all of the defending bots, they will just get stuck at the edge of the arena, Now I'm just stuck with my bots doing nothing. There's a percentage counter counting up, but I don't know what it does, and it's pretty slow.

**Expected outcome**
The phase should end and I should have the opportunity to place more bots.

**Actual outcome**
Bots get stuck at the edge of the arena.

**Bowser information**
{"name":"Chrome","chrome":true,"version":"56.0","blink":true,"windows":true,"a":true}

**In-game information**
Player ID: 147
Player name: Dag

**Comment by Adam13531** [comment]
This **is** a bug, but there are two components to it:

1. Having the ability to skip the current phase without having to watch a replay wasn't originally planned. I'm still not sure if I want to add it in. What happened in your case is that your bots were alive but weren't doing anything, so the replay data generated probably looked something like this:
```
Time 0: move right
Time 1: fire artillery
Time 2-100: a bunch of other valid stuff involving moving/killing; you saw the progress increase during this time
Time 3000: end game
```

Given that there's a bunch of time between 100 and 3000 with no visible actions occurring, the progress bar won't increase, but the game also technically wasn't over as your bots were just doing nothing during that time.

2. The actual bug is that the progress bar doesn't reflect the amount of **time** that passed, it reflects the amount of **turns** that have passed. No turns were recorded from 100-3000 that produced any actions, so the progress bar just stopped.

I just wanted to add some information for now. I'm not totally sure how I'm going to address this yet.

**Comment by Adam13531** [comment]
I fixed this by basing replay progress on **time** instead of **turns** as I mentioned in point number 2 above.

Note that if you have dead time in your replay, you'll still see nothing happening in the Arena itself, but at least you'll see the progress bar get updated.

The fix will be present beyond version 0.1.21. Commit hash: d5506720511e2acdf12257995385a6397b7d41d4

**Comment by mlindgren** [comment]
This didn't actually happen while I was watching a replay. I was actively attacking, but I had to sit and wait for a long time because my bots weren't doing anything. Not sure if that changes your analysis.

Sent from my phone

On Mar 28, 2017, at 16:21, Adam <notifications@github.com<mailto:notifications@github.com>> wrote:


I fixed this by basing replay progress on time instead of turns as I mentioned in point number 2 above.

Note that if you have dead time in your replay, you'll still see nothing happening in the Arena itself, but at least you'll see the progress bar get updated.

The fix will be present beyond version 0.1.21. Commit hash: d5506720511e2acdf12257995385a6397b7d41d4
